### PR TITLE
[ContactStructuralMechanicsApplication] Fixing two tests and removing subrange

### PR DESCRIPTION
**Pull request by loumalouomega**
The subrange is removed toward AMatrix

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp [hunk 1]
```diff
@@ -254,11 +254,13 @@ namespace Kratos
         KRATOS_TEST_CASE_IN_SUITE(WeightedGap1, KratosContactStructuralMechanicsFastSuite)
         {
             Model this_model;
-            ModelPart& r_model_part = this_model.CreateModelPart("Contact", 2);
+            ModelPart& r_model_part = this_model.CreateModelPart("Contact", 3);
             r_model_part.CreateSubModelPart("ComputingContact");
 
             r_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
             r_model_part.AddNodalSolutionStepVariable(WEIGHTED_GAP);
+            r_model_part.AddNodalSolutionStepVariable(WEIGHTED_SLIP);
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp:262]: does this have to be historical? ;)

File: applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_mortar_contact_condition.cpp [hunk 2]
```diff
@@ -343,12 +343,16 @@ void AugmentedLagrangianMethodMortarContactCondition<TDim,TNumNodes,TFrictional,
...
+            for (IndexType i_dim = 0; i_dim < TDim; ++i_dim) {
+                aux_normal[i_dim] = normal[i_dim];
+            }
             const array_1d<double, TDim> aux_array = row(D_x1_M_x2, i_node);
 
             double& weighted_gap = slave_geometry[i_node].FastGetSolutionStepValue(WEIGHTED_GAP);
 
             #pragma omp atomic
-            weighted_gap += inner_prod(aux_array, - subrange(normal, 0, TDim));
+            weighted_gap += inner_prod(aux_array, - aux_normal);
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_mortar_contact_condition.cpp:355]: subrange is not supported by AMatrix right?

**Review by philbucher — APPROVED** [review]
ok

File: applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp:262]: The slip for sure

Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp:262]: and the gap for a derivative too

File: applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_mortar_contact_condition.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_mortar_contact_condition.cpp:355]: Nope

**Pull request merged**